azure-pipelines.yml

```diff
@@ -0,0 +1,112 @@
+
+          $freememInMB = ((Get-CimInstance -Class Win32_OperatingSystem).FreePhysicalMemory / 1024)
+          $freememInMBRounded = [Math]::Round($freememInMB)
+          if ($freememInMbRounded -lt 4096)
+          {
+            Write-Error "The host does not have enough resources to install and run AksEdge"
+            Exit 1
+          }
+          Write-Host "The host has $freememInMBRounded free memory, AksEdge can be installed on it"
+
+          $PSConfiguration = Get-ExecutionPolicy
+          if ($PSConfiguration -ne "Bypass" -and $PSConfiguration -ne "Unrestricted")
+          {
+            Write-Error "The host current powershell configuration is $PSConfiguration, expected configuration is Bypass or Unrestricted"
+            Exit 1
+          }
+          Write-Host "The host current powershell configuration is $PSConfiguration"
+
+          $SSHCheck = (Get-WindowsCapability -Online | Where-Object Name -like 'OpenSSH.Client*').State
+          $HyperVHyperVisor = (Get-WindowsOptionalFeature -FeatureName Microsoft-Hyper-V-Hypervisor -Online).State
+          $HyperV = (Get-WindowsOptionalFeature -FeatureName Microsoft-Hyper-V -Online).State
+          $HyperVMngPowershell = (Get-WindowsOptionalFeature -FeatureName Microsoft-Hyper-V-Management-PowerShell -Online).State
+          if ($SSHCheck -ne "Installed" -Or $HyperVHyperVisor -ne "Enabled" -Or $HyperV -ne "Enabled" -Or $HyperVMngPowershell -ne "Enabled")
+          {
+            Write-Error "Not all software modules are installed, the software modules that are needed are: OpenSSH.Client, Microsoft-Hyper-V-Hypervisor, Microsoft-Hyper-V, Microsoft-Hyper-V-Management-PowerShell."
+            Exit 1
+          }
+          Write-Host "All the software modules are installed: OpenSSH.Client, Microsoft-Hyper-V-Hypervisor, Microsoft-Hyper-V, Microsoft-Hyper-V-Management-PowerShell."
+        displayName: 'Validating agent state'
+      
+      - powershell: |
+          Install-PackageProvider -Name NuGet -MinimumVersion '2.8.5.201' -Force
+          $modules = Get-Module -ListAvailable
+          Write-Host "Modules available are"
+          Write-Host $modules
+          $psgallery = Get-PSRepository | Where-Object { $_.Name -like "PSGallery" }
+          if ($psgallery.InstallationPolicy -ine "Trusted") {
+              # Do this always as by default PSGallery is untrusted. 
+              # See alternate means to force install rather than making this trusted.
+              Write-Host "Setting PSGallery as Trusted Source"
+              Set-PSRepository -Name "PSGallery" -InstallationPolicy Trusted
+          }
+          else { Write-Host "PSGallery is trusted" -ForegroundColor Green }
+        displayName: 'Enabling PS Reposiotory access to install required module'
+
+      - powershell: |
+          $LogFile = 'e2e_junit.xml'
+          $TestPath = "$(Agent.BuildDirectory)\self\tests"
+          cd $TestPath
+          $AzureConfig = @{
```

Comment:
Is there any rsik having this as part of our GitHub?

Reply:
I don't think it should be because the service principal details are hidden. But if required I can set the subscription details as secret as well.

Reply:
Yes, probably let's try to get everything as secret as possible.